microservices/webapp/vite-react/src/routes/Rooms.tsx

```diff
@@ -0,0 +1,247 @@
+import { NativeSelect, Table } from "@mantine/core"
+import { useState } from 'react';
+import { MultiSelect } from '@mantine/core';
+import { TextInput } from '@mantine/core';
+import { Room } from "../entities/Room";
+
+
+interface paramsObj {
+  name: string;
+  location: string;
+  description: string;
+  [key:string]: string;
+}
+
+
+const paramsObj: paramsObj = {
+  "name": "",
+  "location": "",
+  "description": "",
+  "type": ""
+}; 
+
+function searchAPI(searchParams:URLSearchParams) {
+  const rooms = [
```

Comment:
This schema deviates a tiny bit from what is specified in `api/schemas/items/...`. It's def OK, but be sure to review those to see what our schemas are (and if they need tweaking)